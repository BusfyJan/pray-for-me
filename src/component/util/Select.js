import React from 'react'
import PropTypes from 'prop-types'
import Switch from '@material-ui/core/Switch'
import { withStyles } from '@material-ui/core/styles'

const styles = theme => ({
  iOSSwitchBase: {
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
      easing: theme.transitions.easing.sharp
    })
  },

  iOSBar: {
    borderRadius: 13,
    width: 42,
    height: 26,
    marginTop: -13,
    marginLeft: -21,
    border: 'solid 1px',
    borderColor: theme.palette.grey[400],
    backgroundColor: theme.palette.grey[50],
    opacity: 1,
    transition: theme.transitions.create(['background-color', 'border'])
  },
  iOSIcon: {
    width: 24,
    height: 24
  },
  iOSIconChecked: {
    boxShadow: theme.shadows[1]
  }
})

const Switcher = ({ isChecked, switchClicked, classes }) => (
  <Switch
    disableRipple
    checked={isChecked}
    onChange={switchClicked}
    classes={{
      bar: classes.iOSBar,
      icon: classes.iOSIcon,
      checked: classes.iOSChecked
    }}
  />
)

Switcher.propTypes = {
  isChecked: PropTypes.bool.isRequired,
  switchClicked: PropTypes.func.isRequired
}

export default withStyles(styles)(Switcher)
