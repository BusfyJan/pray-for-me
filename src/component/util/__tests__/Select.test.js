import React from 'react'
import Select from '../Select'
import renderer from 'react-test-renderer'

const mockCallBack = jest.fn()

test('Select render', () => {
  const component = renderer.create(
    <Select isChecked switchClicked={mockCallBack} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
