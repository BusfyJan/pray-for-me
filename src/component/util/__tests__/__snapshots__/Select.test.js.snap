// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render 1`] = `
<span
  className="MuiSwitch-root-5"
>
  <span
    className="MuiButtonBase-root-25 MuiIconButton-root-16 MuiSwitchBase-root-11 MuiSwitchBase-default-13 MuiSwitch-default-8 MuiSwitchBase-checked-14 MuiSwitch-checked-9"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role={undefined}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-22"
    >
      <span
        className="MuiSwitch-icon-7 Switcher-iOSIcon-3"
      />
      <input
        checked={true}
        className="MuiSwitchBase-input-12"
        disabled={undefined}
        name={undefined}
        onChange={[Function]}
        tabIndex={undefined}
        type="checkbox"
        value={undefined}
      />
    </span>
  </span>
  <span
    className="MuiSwitch-bar-6 Switcher-iOSBar-2"
  />
</span>
`;
